### ISSUE TITLE
Different Ram Usage ?

### ISSUE THREAD

**Issue by Tahinli** [ClementTsang/bottom#1297]
### Checklist

- [X] I've looked through [troubleshooting](https://clementtsang.github.io/bottom/nightly/troubleshooting), [known problems](https://clementtsang.github.io/bottom/nightly/support/official/#known-problems), and  [existing open issues](https://github.com/ClementTsang/bottom/issues?q=is%3Aopen+is%3Aissue) for similar issues.

### What operating system and version are you using?

Linux Mint 21.2 Cinnamon

### What architecture are you using?

x86_64 / AMD64

### What terminal(s) are you running bottom on?

alacritty

### What version of bottom are you running?

0.9.6

### How did you install bottom?

Cargo

### Describe the issue

I don't know which one is right, but htop and bottom show different ram usages.

![Screenshot from 2023-09-11 02-30-56](https://github.com/ClementTsang/bottom/assets/96421894/095dae76-6d02-480e-85a9-a6b930fc81d4)

### What is the expected behaviour?

They should be same.

### What is the actual behaviour?

They are not same.

### How can we reproduce this?

I don't know

**Comment by ClementTsang** [comment]
Hm, in htop, would it be possible for you to enable the `Memory [Text]` meter?

But yeah that's weird, it looks fine if I bring them both up on my machine, in that the two match fine:

![image](https://github.com/ClementTsang/bottom/assets/34804052/963c95cc-d058-4edd-8b13-219a062aad4f)

I'll take a look though.

**Comment by ClementTsang** [comment]
Also could you run `free -mh`, and paste the results? Thanks!

**Comment by ClementTsang** [comment]
Hm, was just checking around, and noticed that `sysinfo` just patched a bug around memory retrieval - let me bump the dependency and get a build out.

**Comment by Tahinli** [comment]
Here text
![Screenshot from 2023-09-11 02-46-10](https://github.com/ClementTsang/bottom/assets/96421894/813aa73e-37c0-4902-8176-7a25cd869a02)

here free -mh 
![Screenshot from 2023-09-11 02-47-03](https://github.com/ClementTsang/bottom/assets/96421894/db9aa952-fb95-4945-9841-1f23df6cd36c)

and here linux mint system monitor

![Screenshot from 2023-09-11 02-47-20](https://github.com/ClementTsang/bottom/assets/96421894/373822e9-b02c-4016-b0a5-1856bdae60fa)

**Comment by Tahinli** [comment]
> Hm, was just checking around, and noticed that `sysinfo` just patched a bug around memory retrieval - let me bump the dependency and get a build out.

yeah should be something like this because my computer doesn't make me feel like i'm using 26gb ram lol.
I know what kind of programs are working and potentially how much ram should be consumed.

there should be something because even free -mh says I'm using over 20gb

**Comment by ClementTsang** [comment]
Thanks! And yeah, it... actually looks like `btm` matches closer to what `free` and the system monitor report, so I'm not sure if that means `htop` is calculating things differently here...

I do notice you're running VMs, I wonder if that's causing something weird to happen. But yeah let me get that sysinfo bump out and run a nightly build.

**Comment by Tahinli** [comment]
> Thanks! And yeah, it... actually looks like `btm` matches closer to what `free` and the system monitor report, so I'm not sure if that means `htop` is calculating things differently here...
> 
> I do notice you're running VMs, I wonder if that's causing something weird to happen. But yeah let me get that sysinfo bump out and run a nightly build.

btw i closed vm and tried again. still same. and actually even if it running i just gave it to 2gb ram

**Comment by ClementTsang** [comment]
Thanks, that rules that out at least. Some other questions:

- Could you send a screenshot of `top`?
- Are you using ZFS?

**Comment by Tahinli** [comment]
I killed vm and took screen shot
![Screenshot from 2023-09-11 02-59-03](https://github.com/ClementTsang/bottom/assets/96421894/3057ede4-625a-4fb7-a255-53b753d95630)

if i'm not wrong, yes i'm using zfs
![Screenshot from 2023-09-11 03-01-36](https://github.com/ClementTsang/bottom/assets/96421894/9ccdc4b9-8f8d-4ef9-9cf2-223b8d352ee6)

**Comment by ClementTsang** [comment]
Ah, yeah, that explains it - tools like `free` do not remove the ARC usage from total memory usage, while tools like `htop` do. This matches your above screenshot - in bottom, 19.76 GiB - 15.5 GiB = 4.26 GiB which is about what `htop` is showing.

If you don't mind, I can turn this report to a feature request to add a config option to allow you to enable/disable this. Wouldn't be too hard.

**Comment by Tahinli** [comment]
> Ah, yeah, that explains it - tools like `free` do not remove the ARC (ZFS cache) usage from total memory usage, while tools like `htop` do. This matches your above screenshot - in bottom, 19.76 GiB - 15.5 GiB = 4.26 GiB which is about what `htop` is showing.
> 
> If you don't mind, I can turn this report to a feature request to add a config option to allow you to enable/disable this. Wouldn't be too hard.

yeah sure, actually this makes me happy.

**Comment by Tahinli** [comment]
but as a question, technically which one is right. included zfs cache or not ?

**Comment by ClementTsang** [comment]
They're... well, kinda both right.

And from what I remember (my understanding of ZFS is only rudimentary though!) if memory is needed by the rest of the system then the memory used by the ZFS cache ends up getting reclaimed as needed, so it shouldn't affect your system.

**Comment by jamartin9** [comment]
> I mean if countinue to allocate memory with some heavy programs, am i going to be allowed or my system going to crash / kill process

> if memory is needed by the rest of the system then the memory used by the ZFS cache ends up getting reclaimed as needed, so it shouldn't affect your system.

OOM killer may be invoked if ARC does not free memory fast enough. If your workload allocates a lot of RAM quickly (vms/games/etc) some zfs properties like `zfs_arc_sys_free`/`zfs_arc_max`/etc should probably be configured.

**Comment by duskmoss** [comment]
this would be a nice feature if you get time to work on it :)

**Comment by ClementTsang** [comment]
Seems like this is closed by #1812.
